src/app/workspace/workspace-chooser/workspace-chooser.component.ts

```diff
@@ -72,37 +79,94 @@ export class WorkspaceChooserComponent implements OnInit {
   }
 
   public ngOnInit() {
-    this.store.select(selectWorkspace).pipe(
-      tap(workspace => this.workspace = workspace),
-      switchMap(() => this.organizationService.getOrganizations())
-    ).subscribe(organizations => {
-      this.organizations = organizations;
-
-      if (this.workspace.organizationCode) {
-        const ix: number = this.organizations.findIndex(org =>
-          org.code === this.workspace.organizationCode
-        );
-        if (ix >= 0) {
-          this.activeOrgIx = ix;
-
-          const activeOrganization = this.organizations[this.activeOrgIx];
-          activeOrganization.projects = [];
-          this.projectService.getProjects(activeOrganization.code)
-            .subscribe((projects: Project[]) => {
-              activeOrganization.projects = projects;
-
-              if (this.workspace.projectCode) {
-                const ixProj: number = activeOrganization.projects.findIndex(proj =>
-                  proj.code === this.workspace.projectCode
-                );
-                if (ixProj >= 0) {
-                  this.activeProjIx = ixProj;
+    Observable.combineLatest(
+      this.store.select(selectWorkspace),
+      this.store.select(selectOrganizationsState),
+      this.store.select(selectProjectsState)
+    ).pipe(
+      first()
+    ).subscribe(([workspace, organizationsState, projectsState]) => {
+      this.workspace = workspace;
+      this.organizationsState = organizationsState;
+      this.projectsState = projectsState;
+
+      this.getOrganizations();
+    });
+  }
+
+  private getOrganizations(): void {
+    this.organizationService.getOrganizations().subscribe(
```

Comment:
If you want to rewrite this component using NgRx, you must not call REST services directly. The way of accomplishing the same thing in NgRx is to dispatch an action getting organizations (and projects) from the server and then subscribe to the content of the store in order to show those objects in the UI.

Reply:
But maybe for now you can keep it like it is and address this in the other pull-request.